src/storage/txn/actions/acquire_pessimistic_lock.rs

```diff
@@ -36,13 +36,15 @@ pub fn acquire_pessimistic_lock<S: Snapshot>(
     need_check_existence: bool,
     min_commit_ts: TimeStamp,
     need_old_value: bool,
+    lock_only_if_exists: bool,
 ) -> MvccResult<(Option<Value>, OldValue)> {
     fail_point!("acquire_pessimistic_lock", |err| Err(
         crate::storage::mvcc::txn::make_txn_error(err, &key, reader.start_ts).into()
     ));
-
-    // Update max_ts for Insert operation to guarantee linearizability and snapshot
-    // isolation
+    if lock_only_if_exists {
+        assert!(need_value, "when acquire pessimistic lock, lock_only_if_exists is used, but return_value is not");
```

Comment:
Don't assert this. TiKV should not panic even if the client passes an unreasonable request.